### PR TITLE
Replace GBDT by HGBT and add wrap-up table

### LINKED ISSUE #448: Add wrap-up table for ensemble methods
Adding a wrap-up table to summarize the differences and similarities between bagging vs. boosting methods (ways of training, combining and computation times) may help setting down ideas and improve the success rate of Quiz M6.3 Q1 (which is currently below 70%).

It could be at the end of the _Ensemble based on boosting_ lectures, i.e., inside the [ensemble_hist_gradient_boosting notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/ensemble_hist_gradient_boosting.html).

What do you think?

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
Fixes #448.

This PR reworks the [Hyperparameter tuning for ensemble methods notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/ensemble_hyperparameters.html) to:

- replace GBDT by HGBT (and adapt discussion)
- add wrap-up table
- prefer verbs in present mode

### CODE REVIEW

File: python_scripts/ensemble_hyperparameters.py [hunk 1]
```diff
@@ -136,48 +131,52 @@
...
 # `learning_rate`, and `max_depth` or `max_leaf_nodes` (as previously discussed
 # random forest).
 #
-# Let's first discuss the `max_depth` (or `max_leaf_nodes`) parameter. We saw in
-# the section on gradient-boosting that the algorithm fits the error of the
-# previous tree in the ensemble. Thus, fitting fully grown trees would be
+# Let's first discuss the `max_iter`, which similarly to the `n_estimators`
+# hyperparameter in random forest controls the number of trees in the estimator.
+# The difference is that the actual number of trees required by the model is
+# not directly set by the user but depends on the stopping criteria.
```
Comment (ogrisel) [inline python_scripts/ensemble_hyperparameters.py:144]:
```suggestion
# The difference is that the actual number of trees trained by the model is
# not entirely set by the user but depends also on the stopping criteria: the number of trees can be lower than `max_iter` if
# adding a new tree does not improve the training loss enough.
```

File: python_scripts/ensemble_hyperparameters.py [hunk 2]
```diff
@@ -136,48 +131,52 @@
...
 # have a low depth, typically between 3 to 8 levels, or few leaves ($2^3=8$ to
-# $2^8=256$). Having very weak learners at each step will help reducing
-# overfitting.
+# $2^8=256$). Having very weak learners at each step helps reducing overfitting.
 #
 # With this consideration in mind, the deeper the trees, the faster the
-# residuals will be corrected and less learners are required. Therefore,
-# `n_estimators` should be increased if `max_depth` is lower.
+# residuals are corrected and then less learners are required. Therefore,
+# `max_iter` should be increased if `max_depth` is lower.
```
Comment (ogrisel) [inline python_scripts/ensemble_hyperparameters.py:157]:
```suggestion
# residuals are corrected and then less learners are required. Therefore,
# it can be beneficial to increase `max_iter` if `max_depth` is lower.
```

File: python_scripts/ensemble_hyperparameters.py [hunk 3]
```diff
@@ -196,27 +195,36 @@
 # %% [markdown]
 #
 # ```{caution}
-# Here, we tune the `n_estimators` but be aware that is better to use
-# `early_stopping` as done in the Exercise M6.04.
+# Here, we tune the `max_iter` but be aware that it is better to use
+# `early_stopping` as we will do in the Exercise M6.04.
```
Comment (ogrisel) [inline python_scripts/ensemble_hyperparameters.py:199]:
```suggestion
# Here, we tune the `max_iter` but be aware that it is better set
# `max_iter` to a fixed, large enough value and use parameters linked to
# `early_stopping` as we will do in Exercise M6.04.
```

**Review by ogrisel — APPROVED** [review]
Thanks! A few suggestions but otherwise LGTM!

File: python_scripts/ensemble_hyperparameters.py [hunk 4]
```diff
@@ -196,27 +195,36 @@
...
 
 # %% [markdown]
 # The mean test score in the held-out test set is slightly better than the score
 # of the best model. The reason is that the final model is refitted on the whole
 # training set and therefore, on more data than the cross-validated models of
 # the grid search procedure.
+#
+# We summarize these details in the following table:
+#
+# | **Bagging**                                      | **Boosting**                                       |
```
Comment (ogrisel) [inline python_scripts/ensemble_hyperparameters.py:224]:
```suggestion
# | **Bagging & Random Forests**                     | **Boosting**                                       |
```

File: python_scripts/ensemble_hyperparameters.py [hunk 5]
```diff
@@ -136,48 +131,52 @@
...
+# deep trees.
 
 # %%
 from scipy.stats import loguniform
-from sklearn.ensemble import GradientBoostingRegressor
+from sklearn.ensemble import HistGradientBoostingRegressor
 
 param_distributions = {
-    "n_estimators": [1, 2, 5, 10, 20, 50, 100, 200, 500],
+    "max_iter": [2, 5, 10, 20, 50, 100, 200, 500],
```
Comment (ogrisel) [inline python_scripts/ensemble_hyperparameters.py:174]: Maybe we can try a coarser grid but going to a larger maximum to see if the best model can reach this bound or not.

```suggestion
    "max_iter": [3, 10, 30, 100, 300, 1000],
```
